packages/cookie/README.md

```diff
@@ -0,0 +1,253 @@
+
+// Objects
+await preferencesCookie.serialize({
+  theme: 'dark',
+  language: 'en-US',
+  notifications: true,
+})
+
+// Booleans
+await preferencesCookie.serialize(false)
+
+// Numbers
+await preferencesCookie.serialize(42)
+```
+
+## Cookie Configuration
+
+Cookies can be configured with comprehensive options:
+
+```ts
+import { Cookie } from '@remix-run/cookie'
+
+let cookie = new Cookie('my-cookie', {
+  // Security options
+  secrets: ['secret1', 'secret2'], // For signing (first used for new cookies)
+  httpOnly: true, // Prevent JavaScript access
+  secure: true, // Require HTTPS
+
+  // Scope options
+  domain: '.example.com', // Cookie domain
+  path: '/admin', // Cookie path
+
+  // Expiration options
+  maxAge: 60 * 60 * 24, // Max age in seconds
+  expires: new Date('2025-12-31'), // Absolute expiration date
+
+  // SameSite options
+  sameSite: 'strict', // 'strict' | 'lax' | 'none'
+
+  // Encoding options (from 'cookie' package)
+  encode: (value) => encodeURIComponent(value),
+  decode: (value) => decodeURIComponent(value),
+})
+```
+
+## Secret Rotation
+
+One of the key features is seamless secret rotation for signed cookies:
+
+```ts
+// Start with an initial secret
+let cookie = new Cookie('session', {
+  secrets: ['secret-v1'],
+})
+
+let setCookie1 = await cookie.serialize({ user: 'alice' })
+
+// Later, rotate to a new secret while keeping the old one
+cookie = new Cookie('session', {
+  secrets: ['secret-v2', 'secret-v1'], // New secret first, old ones after
+})
+
+// New cookies use the new secret
+let setCookie2 = await cookie.serialize({ user: 'bob' })
+
+// But old cookies still work
+let oldValue = await cookie.parse(setCookie1.split(';')[0])
+console.log(oldValue) // { user: 'alice' } - still works!
+
+let newValue = await cookie.parse(setCookie2.split(';')[0])
+console.log(newValue) // { user: 'bob' }
+```
+
+## Advanced Usage
+
+### Custom Serialization Options
+
+You can override cookie options when serializing:
+
+```ts
+let cookie = new Cookie('flexible', {
+  maxAge: 60 * 60, // Default 1 hour
+})
+
+// Override for a specific use case
+let longLivedCookie = await cookie.serialize('remember-me', {
+  maxAge: 60 * 60 * 24 * 365, // 1 year
+})
+
+let sessionCookie = await cookie.serialize('temp-data', {
+  maxAge: undefined, // Session cookie (no expiration)
+  secure: false, // Maybe for development
+})
+```
+
+### Error Handling
+
+The library handles various error scenarios gracefully:
+
+```ts
+let cookie = new Cookie('test')
+
+// Missing or malformed cookie headers return null
+await cookie.parse(null) // null
+await cookie.parse('') // null
+await cookie.parse('other=value') // null
+
+// Malformed cookie values return empty object or null
+await cookie.parse('test=invalid-base64@#$') // {}
+
+// Signed cookies with bad signatures return null
+let signedCookie = new Cookie('signed', { secrets: ['secret'] })
+await signedCookie.parse('signed=value.badsignature') // null
+```
+
+## API Reference
```

Comment:
We can leave this section out of the README. We will work on auto-generating API docs for everything here at some point soon.